ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/CommandRunner.java

```diff
@@ -437,6 +460,13 @@ public void run() {
         if (!closed) {
           throw wue;
         }
+      } catch (final OffsetOutOfRangeException e) {
+        LOG.warn("The command topic offset was reset. CommandRunner thread exiting.");
+        state = new Status(
+            CommandRunnerStatus.DEGRADED,
+            CommandRunnerDegradedReason.COMMAND_TOPIC_DELETED
```

Comment:
does offset reset always mean command topic deleted? it could mean the retention was configured incorrectly perhaps?

Reply:
Every time we start up the server, we'd check/override the command topic properties such that the `retention.ms=-1` although I guess it is possible for the user themselves to override the retention config on the command topic which could trigger this. I can enhance the error message so that it mentions the degraded state is a result of `the command topic being deleted or the configurations being modified for it`

The REASON can be updated to `COMMAND_TOPIC_DELETED_OR_MODIFIED`